### PR TITLE
add github-guidelines.md

### PR DESCRIPTION
**Pull request by khareyash05**
With reference to issue #139 
Added github-guidelines.md
It is a simple file which even beginners can use to contribute to open source
Added the following
1.Difference between GIT and GITHUB
2. How to clone,fork repository
3. How to create a branch and then use git push to push to repo
4. Create a PR
5. Squash commits in a single issue into one
6. Updating the forked and local repo as the updations are made in the upstream
@garg3133 have a look

### CODE REVIEW

File: github-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+## GIT AND GITHUB
+***
+Before continuing we want to clarify the difference between Git and Github. Git is a version control system(VCS) which is a tool to manage the history of our Source Code. GitHub is a hosting service for Git projects.
+
+We assume you have created an account on Github and installed Git on your System.
+
+Now tell Git your name and E-mail (used on Github) address.
+
+``` $ git config --global user.name "YOUR NAME" ```
+```$ git config --global user.email "YOUR EMAIL ADDRESS"``` <br>
```
Comment (garg3133) [inline github-guidelines.md:10]: This is not a way to write single line code. Either enclose each command inside a single backtick or enclose both commands inside three backticks like this:

\`\`\`
$ git config --global user.name "YOUR NAME"
$ git config --global user.email "YOUR EMAIL ADDRESS"
\`\`\`

Also, don't use `<br>`, instead just add two whitespaces at the end of the line.

File: github-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+## GIT AND GITHUB
+***
```
Comment (garg3133) [inline github-guidelines.md:2]: Don't put these extra lines below each heading. Markdown already provides a line below headings.

File: github-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+You will see the benefits of adding remote later.
+
+### SYNCHRONIZING YOUR FORK -
+***
+Open Source projects have a number of contributors who can push code anytime. So it is necessary to make your forked copy equal with the original repository. The remote added above called Upstream helps in this.
+
+```$ git checkout master```
+```$ git fetch upstream```
+```$ git merge upstream/master```
+```$ git push origin master``` <br>
```
Comment (garg3133) [inline github-guidelines.md:49]: Same thing applies here and at all other places.

File: github-guidelines.md (hunk 1 above)
Comment (khareyash05) [reply github-guidelines.md:10]: I will do that

File: github-guidelines.md (hunk 2 above)
Comment (khareyash05) [reply github-guidelines.md:2]: Okay

**Comment by khareyash05** [comment]
Creating a new PR give me few mins

**Pull request closed without merging**